### PR TITLE
add already enrolled warning on plan confirmation

### PR DESCRIPTION
**Pull request by jayreddy519**
[ME-181147353](https://www.pivotaltracker.com/n/projects/2502930/stories/181147353)

### CODE REVIEW

File: app/domain/operations/individual/fetch_existing_coverage.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        family_ids = []
+        enrollment_members.each do |enr_member|
+          person = enr_member.person
+          family_ids << Family.find_all_by_person(person).pluck(:id) if person.present?
+        end
+        family_ids.count > 0 ? Success(family_ids.uniq.flatten) : Failure('No families found for members')
+      end
+
+      def prior_enrollments(family_ids, enrollment)
+        all_enrollments = HbxEnrollment.where({:family_id.in => family_ids, :aasm_state => {"$in" => HbxEnrollment::ENROLLED_STATUSES + HbxEnrollment::RENEWAL_STATUSES }, coverage_kind:  enrollment.first.coverage_kind, kind:  enrollment.first.kind})
```
Comment (ymhari) [inline app/domain/operations/individual/fetch_existing_coverage.rb:51]: Should the auto-renewal be considered as existing coverage?

File: app/controllers/insured/plan_shoppings_controller.rb [hunk 2]
```diff
@@ -339,6 +339,16 @@ def plans
 
   private
 
+  def dependents_with_existing_coverage(enrollment)
+    existing_coverages = ::Operations::Individual::FetchExistingCoverage.new.call({enrollment_id: enrollment.id})
+    existing_enrollments = existing_coverages.success? ? existing_coverages.value! : []
+    primary_subscriber = enrollment.subscriber
+    existing_dependent_enrollments = existing_enrollments.reject{ |enr| enr.subscriber.applicant_id == primary_subscriber.applicant_id }
+    return [] if existing_dependent_enrollments.nil? || existing_dependent_enrollments.count == 0
```
Comment (ymhari) [inline app/controllers/insured/plan_shoppings_controller.rb:347]: No value is returning an empty array here.

File: app/views/insured/plan_shoppings/_prior_coverage_information.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<div class="alert alert-notice">
+  <a href="#" data-dismiss="alert" ></a>
+  <div class="row">
+    <div class="col-md-11 col-sm-12 col-xs-12">
+        <div class="container">
+        	<h4><%= l10n("coverage_info_msg") %> <%= l10n("site_url") %></h4>
```
Comment (ymhari) [inline app/views/insured/plan_shoppings/_prior_coverage_information.html.erb:6]: This is not a url.

File: config/client_config/dc/system/config/templates/features/enroll_app/enroll_app.yml [hunk 4]
```diff
@@ -690,6 +690,9 @@ registry:
       - key: :custom_exceptions_controller
         item: :custom_exceptions_controller
         is_enabled: false
+      - key: :fetch_existing_coverage
```
Comment (ymhari) [inline config/client_config/dc/system/config/templates/features/enroll_app/enroll_app.yml:693]: Change the key name.

File: app/views/insured/plan_shoppings/thankyou.html.erb [hunk 5]
```diff
@@ -8,6 +8,7 @@
         <div class="col-md-12 col-sm-12 no-pd col-xs-12" id="confirm_plan">
           <h1 class="darkblue"><%= l10n('confirm_your_plan_selection') %></h1>
           <h4><%= l10n('insured.plan_shoppings.thankyou.confirm_your_plan_selection.content') %></h4>
+          <%= render partial: "insured/plan_shoppings/prior_coverage_information", locals: {dependent_members: @dependent_members} if @dependent_members.present? && EnrollRegistry.feature_enabled?(:fetch_existing_coverage) %>
```
Comment (ymhari) [inline app/views/insured/plan_shoppings/thankyou.html.erb:11]: Do we need an RR setting check here?

File: app/domain/operations/individual/fetch_existing_coverage.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module Operations
+  module Individual
```
Comment (ymhari) [inline app/domain/operations/individual/fetch_existing_coverage.rb:7]: This is a generic operation not specific to individual market.

File: spec/domain/operations/individual/fetch_existing_coverage_spec.rb [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                       coverage_start_on: effective_on,
+                       applicant_id: dependents.last.id)
+    end
+    let(:hbx_en_member3) do
+      FactoryBot.build(:hbx_enrollment_member,
+                       eligibility_date: effective_on + 6.months,
+                       coverage_start_on: effective_on + 6.months,
+                       applicant_id: dependents.last.id)
+    end
+    let!(:enrollment1) do
```
Comment (ymhari) [inline spec/domain/operations/individual/fetch_existing_coverage_spec.rb:38]: Do we need 6 enrollment here? Can we mock some if needed?

File: spec/domain/operations/individual/fetch_existing_coverage_spec.rb [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+        enrollment2.save!
+      end
+
+      it "returns failure" do
+        expect(subject).not_to be_success
+        expect(subject.failure).to eq "Enrollment does not include dependents"
+      end
+    end
+
+    context "hbx enrollment members with shopping enrollment" do
```
Comment (ymhari) [inline spec/domain/operations/individual/fetch_existing_coverage_spec.rb:146]: Incorrect context the operation doesn't care that it should not work for the shop market.  Rather this context should be in the controller spec.

File: spec/domain/operations/individual/fetch_existing_coverage_spec.rb [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+      it "returns success" do
+        expect(subject).to be_success
+        expect(subject.success).to eq [enrollment1, enrollment3]
+        # SHOP Enrollment not included
+        expect(subject.success).not_to include [enrollment4]
+      end
+    end
+
```
Comment (ymhari) [inline spec/domain/operations/individual/fetch_existing_coverage_spec.rb:154]: Need more scenarios to check the market_kind, coverage_kind, what is the enrollment as no dependents, and if the enrollments are missing family_ids.

**Review by ymhari — CHANGES_REQUESTED** [review]

**Review by ymhari — APPROVED** [review]
Adds a warning message when dependents are already enrolled in another active enrollment.

**Review by RyanEddyIC — APPROVED** [review]

**Pull request merged**